src/pages/Detail/index.jsx

```diff
@@ -0,0 +1,244 @@
+    const dispatch = useDispatch();
+
+    const products = useSelector(state => state.products);
+    const sizes = useSelector(state => state.sizes);
+    const loading = useSelector(state => state.loading);
+
+    const productId = useSelector(state => state.detailProduct.id);
+    const productName = useSelector(state => state.detailProduct.productName);
+    const categoryId = useSelector(state => state.detailProduct.categoryId);
+    const description = useSelector(state => state.detailProduct.description);
+    const information = useSelector(state => state.detailProduct.information);
+    const newPrice = useSelector(state => state.detailProduct.newPrice);
+    const oldPrice = useSelector(state => state.detailProduct.oldPrice);
+    const totalRating = useSelector(state => state.detailProduct.totalRating);
+    const quantity = useSelector(state => state.detailProduct.quantity);
+    const images = useSelector(state => state.detailProduct.images);
+    const imageShow = useSelector(state => state.detailProduct.imageShow);
+    const relatedProduct = useSelector(state => state.detailProduct.relatedProduct);
+
+    useEffect(() => {
+        const product = products.find(p => p.id === parseInt(id));
+        if (product) {
+            document.title = t(`${product.productName}`);
+            dispatch(actions.actDetailProduct(product));
+            dispatch(actions.actRelatedProduct(products));
+        }
+    }, [products, id]);
+
+    useEffect(() => {
+        dispatch(actions.actShowLoading());
+
+        setTimeout(() => {
+            dispatch(actions.actHideLoading());
+        }, 1000)
+    }, []);
+
+    const changeQuantity = (event) => {
+        const count = parseInt(event.target.value);
+        if (count && count <= 500) {
+            dispatch(actions.actChangeQuantityDetail(count));
+        } else if (count > 500) {
+            toastMsg.fail(`${t('you can not set too 500 product')} !!!`);
+            dispatch(actions.actChangeQuantityDetail(500));
+        } else {
+            dispatch(actions.actChangeQuantityDetail(1));
+        }
+    }
+
+    const clickChangeQuantity = (number) => {
+        dispatch(actions.actChangeQuantityDetail(quantity + number));
+    }
+
+    const changeSize = (event) => {
+        const sizeId = parseInt(event.target.value);
+        dispatch(actions.actChangeSizeDetail(sizeId));
+    }
+
+    const changeImage = (image) => {
+        dispatch(actions.actChangeImageDetail(image));
+    }
+
+    return (
+        <main className="main">
+
+            <Container>
+                <ContentHeader
+                    label={categoryId.replace(/-/g, " ")}
+                    path={`product/${categoryId}`}
+                    label_2={productName}
+                />
+
+                <section className="section__product-des">
+                    <Grid container spacing={3}>
+                        <Grid item xs={12} md={6}>
+                            {loading && <LoadingProduct />}
```

Comment:
Dùng toán tử 3 ngôi thay thế nhé
```suggestion
                            {loading ? <LoadingProduct /> : <div className="block__product-image"></div>}
```